kotlin-eclipse-core/src/org/jetbrains/kotlin/core/preferences/KotlinProperties.kt

```diff
@@ -33,9 +35,37 @@ class KotlinProperties(scope: IScopeContext = InstanceScope.INSTANCE) : Preferen
     }
 
     val compilerPlugins by ChildCollection(::CompilerPlugin)
-    
+
     var compilerFlags by StringPreference()
 
+    val analyzerCompilerFlags: Map<AnalysisFlag<*>, Any?>
+        get() = compilerFlags?.split("\\s+".toRegex())?.mapNotNull { flagString ->
```

Comment:
If you want you can flatten this expression by using series of `map` (with split by `"="`), `filter` and then `mapNotNull` (with `CompilerFlagsMapping.analysisFlagsMapping[...`) instead of big `mapNotNull`.